website/project/model.py

```diff
@@ -2398,18 +2398,24 @@ def set_privacy(self, permissions, auth=None, log=True):
         :param bool log: Whether to add a NodeLog for the privacy change.
         """
         if permissions == 'public' and not self.is_public:
-            if self.is_registration and (self.embargo_end_date or self.pending_embargo):
-                self.embargo.state = Embargo.CANCELLED
-                self.registered_from.add_log(
-                    action=NodeLog.EMBARGO_CANCELLED,
-                    params={
-                        'node': self._id,
-                        'embargo_id': self.embargo._id,
-                    },
-                    auth=auth,
-                )
-                self.embargo.save()
-            self.is_public = True
+            if self.is_registration:
+                def r_set_public(node):
```

Comment:
@sloria - if a private registration had children, this iterates them recursively and sets them to public. This situation can be encountered when a legacy private registration is made public.